### PR TITLE
feat: manage task resources

### PR DESCRIPTION
**Pull request by aaryanpunia**

### CODE REVIEW

File: actix-server/src/data/models.rs [hunk 1]
```diff
@@ -510,3 +537,139 @@ impl Company {
...
+    Queryable,
+    Insertable,
+    Selectable,
+    Clone,
+    ToSchema,
+    Identifiable,
+    Associations,
+)]
+#[schema(example = json!({
+    "id": "taskdeal-b8b8b8b8-b8b8-b8b8-b8b8-b8b8b8b8b8b8",
```
Comment (drew-harris) [inline actix-server/src/data/models.rs:645]: nitpick: prefix in example is wrong

File: actix-server/src/errors.rs [hunk 2]
```diff
@@ -65,6 +67,14 @@ impl From<ParseError> for ServiceError {
...
 }
 
+impl From<PrefixParseError> for ServiceError {
+    fn from(error: PrefixParseError) -> ServiceError {
+        match error {
+            _ => ServiceError::BadRequest("Invalid Prefix".into()),
+        }
+    }
+}
+
```
Comment (drew-harris) [inline actix-server/src/errors.rs:77]: this is awesome

File: actix-server/src/data/models.rs (hunk 1 above)
Comment (aaryanpunia) [reply actix-server/src/data/models.rs]: fixed!

**Comment by drew-harris** [comment]
tested out all the relation stuff for tasks <-> deals, looked the the code for the other links but assume its the same. looks good to me and if theres anything i missed itll get found when we incorporate it to the cli

make sure you rebase the pr when its time to merge

**Review by drew-harris — APPROVED** [review]

**Pull request merged**